patches/net/minecraft/client/gui/Gui.java.patch

```diff
@@ -115,6 +115,23 @@
                  int j;
                  if (this.animateOverlayMessageColor) {
                      j = Mth.hsvToArgb(f / 50.0F, 0.7F, 0.6F, i);
+@@ -323,9 +_,16 @@
+     private void renderChat(GuiGraphics p_316307_, DeltaTracker p_348631_) {
+         if (!this.chat.isChatFocused()) {
+             Window window = this.minecraft.getWindow();
++            // Neo: Allow customizing position of chat component
++            var chatBottomMargin = 40; // See ChatComponent#BOTTOM_MARGIN (used in translate calls in ChatComponent#render)
++            var event = new net.neoforged.neoforge.client.event.CustomizeGuiOverlayEvent.Chat(window, p_316307_, p_348631_, 0, p_316307_.guiHeight() - chatBottomMargin);
++            // The event is given the absolute Y position; account for chat component's own offsetting here
++            p_316307_.pose().pushPose();
++            p_316307_.pose().translate(event.getPosX(), event.getPosY() - (p_316307_.guiHeight() + chatBottomMargin) / this.chat.getScale(), 0.0F);
```

Comment:
Are you sure that it's `+ chatBottomMargin` and not `- chatBottomMargin` as the original (https://github.com/neoforged/NeoForge/blob/8ef74db358eb212361b84b2d671695b983426935/src/main/java/net/neoforged/neoforge/client/gui/overlay/ExtendedGui.java#L422) would suggest?

The original was also scaling `event.getPosY()`.

Reply:
Oh. I made a mistake -- I wrote the parentheses and the scaling division afterwards in that order, and I didn't notice that (a) my parentheses changed the meaning from the original and (b) the division now only applies to the part in parentheses.

I swear I'm an experienced developer. (/jk)